### PR TITLE
I341 demand calculation hq masterplan

### PR DESCRIPTION
**Pull request by daren-thomas**
This branch attempts to solve the bug #341 by adding a fallback to the newton solve.

### CODE REVIEW

File: cea/demand/ventilation_model.py [hunk 1]
```diff
@@ -409,7 +409,7 @@ def allocate_default_leakage_paths(coeff_lea_zone, area_facade_zone, area_roof_z
 
     coeff_lea_path = np.zeros(5)
     height_lea_path = np.zeros(5)
-    orientation_lea_path = np.zeros(5)
+    orientation_lea_path = [0, 1, 0, 1, 2]
```
Comment (JIMENOFONSECA) [inline cea/demand/ventilation_model.py]: @gabriel-happle does this affect the model? or can we do this modification?

File: cea/demand/ventilation_model.py [hunk 2]
```diff
@@ -565,7 +565,7 @@ def allocate_default_ventilation_openings(coeff_vent_zone, height_zone):
     # initialize
     coeff_vent_path = np.zeros(4)
     height_vent_path = np.zeros(4)
-    orientation_vent_path = np.zeros(4)
+    orientation_vent_path = [0, 1, 0, 1]
```
Comment (JIMENOFONSECA) [inline cea/demand/ventilation_model.py]: @gabriel-happle does this affect the model? or can we do this modification?

File: cea/demand/ventilation_model.py (hunk 1 above)
Comment (gabriel-happle) [inline cea/demand/ventilation_model.py]: No. Right now it does not have any implications. As windows are not used anyways in any calculation. As far as I understand.
Why would you change that?
And what does this have to do with cooling coils?

File: cea/demand/ventilation_model.py (hunk 2 above)
Comment (gabriel-happle) [inline cea/demand/ventilation_model.py]: No. Right now it does not have any implications. As windows are not used anyways in any calculation. As far as I understand.
Why would you change that?
And what does this have to do with cooling coils?

File: cea/demand/ventilation_model.py (hunk 1 above)
Comment (JIMENOFONSECA) [inline cea/demand/ventilation_model.py]: @daren-thomas ?

File: cea/demand/ventilation_model.py (hunk 2 above)
Comment (JIMENOFONSECA) [inline cea/demand/ventilation_model.py]: @daren-thomas ?

File: cea/demand/ventilation_model.py (hunk 1 above)
Comment (daren-thomas) [inline cea/demand/ventilation_model.py]: The reason has to do with data types. `np.zeroes` returns an array of floats, but `orientation_lea_path` is actually used to retrieve an index into another array. Indexes need to be integers. I was getting some errors while debugging because they were floats.

BTW: once you have indexes into arrays to retrieve array indexes into other arrays, that can be considered a code smell and yes, here would be a good place to sprinkle some objects...

File: cea/demand/ventilation_model.py (hunk 2 above)
Comment (daren-thomas) [inline cea/demand/ventilation_model.py]: Again. `orientation_vent_path`  is used to retrieve an array index and should be integral. We could also just use `map(int, np.zeros(4))`. But, why not just assign the correct indexes with a list literal? As mentioned above, the code structure still needs some work...

File: cea/technologies/heating_coils.py [hunk 3]
```diff
@@ -8,6 +8,7 @@
 from __future__ import division
 import scipy.optimize as sopt
 import scipy
+import numpy as np
```
Comment (daren-thomas) [inline cea/technologies/heating_coils.py]: this can probably be removed.

File: cea/technologies/heating_coils.py [hunk 4]
```diff
@@ -83,7 +84,15 @@ def fh(x):
...
-        result = sopt.newton(fh, trc0, maxiter=100, tol=0.01) - 273
+        try:
+            result = sopt.newton(fh, trc0, maxiter=100, tol=0.01) - 273
+        except RuntimeError:
+            print('Newton optimization failed, using slower bisect algorithm...')
+            result = sopt.bisect(fh, 0, 350, xtol=0.01, maxiter=500) - 273
+
+
+        if Ta_sup_cs == Ta_re_cs:
+            print 'Ta_sup_cs == Ta_re_cs:', Ta_sup_cs
```
Comment (daren-thomas) [inline cea/technologies/heating_coils.py]: this can be removed: i only added it to count how often that case happens (it happens a lot!)
